controllers/managedClusterResources.go

```diff
@@ -552,23 +553,20 @@ func (r *ClusterGroupUpgradeReconciler) getPrecacheJobTemplateData(
 	return rv, nil
 }
 
-func (r *ClusterGroupUpgradeReconciler) getBackupJobTemplateData(clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, clusterName string) *templateData {
+func (r *ClusterGroupUpgradeReconciler) getBackupJobTemplateData(clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, clusterName string) (*templateData, error) {
 
 	rv := new(templateData)
-
 	rv.Cluster = clusterName
 	rv.JobTimeout = uint64(
 		clusterGroupUpgrade.Spec.RemediationStrategy.Timeout)
 
-	// TODO: currently using hard-coded recovery image, we must get
-	// image from csv.
-	//	image, err := r.getPrecacheimagePullSpec(ctx, clusterGroupUpgrade)
-	//	if err != nil {
-	//		return rv, err
-	//	}
-	//	rv.WorkloadImage = image
-
-	return rv
+	rv.WorkloadImage = os.Getenv("RECOVERY_IMG")
+	r.Log.Info("[getBackupJobTemplateData]", "workload image", rv.WorkloadImage)
+	if rv.WorkloadImage == "" {
+		return rv, fmt.Errorf(
+			"can't find recovery image pull spec in environment")
```

Comment:
IIUC this basically fails if the env var isn't set or empty. How about referencing that in the comment as well? Something along the lines of. RECOVERY_IMG is not ser or empty.

Reply:
Done as suggested with [92d2fb6](https://github.com/openshift-kni/cluster-group-upgrades-operator/commit/92d2fb6f21aa4fd4b8894fc97da028d8a9e4dbf0)